libraries/Microsoft.Bot.Connector/Authentication/IAppId.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace Microsoft.Bot.Connector.Authentication
+{
+    /// <summary>
+    /// Represents the Id of a bot for a credential provider.
+    /// </summary>
+    public interface IAppId
```

Comment:
This seems a little extreme. I mean this interface doesn't really correspond to any real world concept - it's hard to imagine meaningful alternative implementations. Basically its hard to explain, and these constructs (classes, interfaces, abstract classes etc etc) only really exist to explain the code.

This scenario is specific to the BotFrameworkAdapter right? Could we add a property to the BotFrameworkAdapter. I guess I need to recap on the scenario. I'll read the issue.